src/main/java/bio/terra/service/resourcemanagement/ResourceService.java

```diff
@@ -73,13 +81,30 @@ public GoogleBucketResource getOrCreateBucketForFile(Dataset dataset,
             billingProfile,
             null);
 
-        return bucketService.getOrCreateBucket(
-            dataLocationSelector.bucketForFile(datasetName, billingProfile),
-            projectResource,
+        List<UUID> bucketsForDataset = datasetBucketDao.getBucketForDatasetId(UUID.fromString(datasetId));
+        String bucketName = dataLocationSelector.bucketForFile(datasetId, billingProfile);
+
+        List<GoogleBucketResource> bucketsForBillingProfile = bucketsForDataset.stream()
+            .map(bucketUUID -> lookupBucket(bucketUUID.toString()))
+            .filter(bucket -> bucketIsForBillingProfile(bucket, billingProfile))
+        .collect(Collectors.toList());
+
+        if (bucketsForBillingProfile.size() > 0) {
+            GoogleBucketResource bucket = bucketsForBillingProfile.get(0);
+            bucketName = bucket.getName();
```

Comment:
If we reach this point, do we know for sure if a bucket already exists or not? Would it be worth splitting out the case where the bucket exists and the doesn't so `getOrCreateBucket()` is no longer needed, e.g., we'd have `getBucket()` and `createBucket()`?

Reply:
seems like the pattern for most of these methods is `getOrCreateX`, and if i were to split it up i would just have to write the same logic in the places that it's being used so maybe not worth it for now

Reply:
I think we need the "getOrCreateBucket" to handle the different DataLocationSelectors, but with RBS, we can ditch that and split these up!